client/src/components/files/FileOpenFallbackChoice.vue

```diff
@@ -38,17 +40,34 @@
 import { IonButton, IonButtons, modalController } from '@ionic/vue';
 import { isDesktop, isWeb } from '@/parsec';
 import { OpenFallbackChoice } from '@/components/files';
-import { MsModal } from 'megashark-lib';
+import { MsModal, MsModalResult } from 'megashark-lib';
+import { StorageManager, StorageManagerKey } from '@/services/storageManager';
+import { inject } from 'vue';
+import { askDownloadConfirmation } from '@/views/files';
 
 defineProps<{
   viewerOption?: boolean;
 }>();
 
+const storageManager: StorageManager = inject(StorageManagerKey)!;
+
 async function onViewerClick(): Promise<void> {
   await modalController.dismiss(OpenFallbackChoice.View);
 }
 
 async function onDownloadClick(): Promise<void> {
+  const config = await storageManager.retrieveConfig();
```

Comment:
I think you can factorize the whole config thing. It's going to be used multiple times.